TombEngine/Specific/configuration.cpp

```diff
@@ -18,290 +23,35 @@ using namespace TEN::Utils;
-	if (RegCreateKeyExA(rootKey, REGKEY_GAMEPLAY, 0, NULL, REG_OPTION_NON_VOLATILE, KEY_WRITE, NULL, &gameplayKey, NULL) != ERROR_SUCCESS)
-	{
-		RegCloseKey(rootKey);
-		RegCloseKey(graphicsKey);
-		RegCloseKey(soundKey);
-		RegCloseKey(gameplayKey);
-		return false;
-	}
-
-	// Set Gameplay keys.
-	if (SetBoolRegKey(gameplayKey, REGKEY_ENABLE_SUBTITLES, g_Configuration.EnableSubtitles) != ERROR_SUCCESS ||
-		SetBoolRegKey(gameplayKey, REGKEY_ENABLE_AUTO_MONKEY_JUMP, g_Configuration.EnableAutoMonkeySwingJump) != ERROR_SUCCESS ||
-		SetBoolRegKey(gameplayKey, REGKEY_ENABLE_AUTO_TARGETING, g_Configuration.EnableAutoTargeting) != ERROR_SUCCESS ||
-		SetBoolRegKey(gameplayKey, REGKEY_ENABLE_TARGET_HIGHLIGHTER, g_Configuration.EnableTargetHighlighter) != ERROR_SUCCESS ||
-		SetBoolRegKey(gameplayKey, REGKEY_ENABLE_INTERACTION_HIGHLIGHTER, g_Configuration.EnableInteractionHighlighter) != ERROR_SUCCESS ||
-		SetBoolRegKey(gameplayKey, REGKEY_ENABLE_RUMBLE, g_Configuration.EnableRumble) != ERROR_SUCCESS ||
-		SetBoolRegKey(gameplayKey, REGKEY_ENABLE_THUMBSTICK_CAMERA, g_Configuration.EnableThumbstickCamera) != ERROR_SUCCESS)
-	{
-		RegCloseKey(rootKey);
-		RegCloseKey(graphicsKey);
-		RegCloseKey(soundKey);
-		RegCloseKey(gameplayKey);
-		return false;
-	}
-
-	// Open Input subkey.
-	HKEY inputKey = NULL;
-	if (RegCreateKeyExA(rootKey, REGKEY_INPUT, 0, NULL, REG_OPTION_NON_VOLATILE, KEY_WRITE, NULL, &inputKey, NULL) != ERROR_SUCCESS)
-	{
-		RegCloseKey(rootKey);
-		RegCloseKey(graphicsKey);
-		RegCloseKey(soundKey);
-		RegCloseKey(gameplayKey);
-		RegCloseKey(inputKey);
-		return false;
-	}
-
-	// Set Input keys.
-	if (SetDWORDRegKey(inputKey, REGKEY_MOUSE_SENSITIVITY, g_Configuration.MouseSensitivity) != ERROR_SUCCESS ||
-		SetDWORDRegKey(inputKey, REGKEY_MENU_OPTION_LOOPING_MODE, (int)g_Configuration.MenuOptionLoopingMode) != ERROR_SUCCESS)
-	{
-		RegCloseKey(rootKey);
-		RegCloseKey(graphicsKey);
-		RegCloseKey(soundKey);
-		RegCloseKey(gameplayKey);
-		RegCloseKey(inputKey);
-		return false;
-	}
-
-	if (g_Configuration.Bindings.empty())
-		g_Configuration.Bindings = DEFAULT_KEYBOARD_MOUSE_BINDING_PROFILE;
-
-	// Set Input binding keys.
-	for (auto actionGroupID : USER_ACTION_GROUP_IDS)
-	{
-		const auto& actionIDGroup = ACTION_ID_GROUPS[(int)actionGroupID];
-		for (auto actionID : actionIDGroup)
-		{
-			char buffer[9];
-			sprintf(buffer, "Action%d", (int)actionID);
-
-			if (SetDWORDRegKey(inputKey, buffer, g_Configuration.Bindings.at(actionID)) != ERROR_SUCCESS)
-			{
-				RegCloseKey(rootKey);
-				RegCloseKey(graphicsKey);
-				RegCloseKey(soundKey);
-				RegCloseKey(gameplayKey);
-				RegCloseKey(inputKey);
-				return false;
-			}
-		}
-	}
-
-	// Close registry keys.
-	RegCloseKey(rootKey);
-	RegCloseKey(graphicsKey);
-	RegCloseKey(soundKey);
-	RegCloseKey(gameplayKey);
-	RegCloseKey(inputKey);
-	return true;
+	SDL_IOStream* rw = SDL_IOFromFile(path.c_str(), "rb");
+	if (!rw) return false;
+	Sint64 size = SDL_GetIOSize(rw);
```

Comment:
Use types according to coding conventions, please.

Reply:
Done